scene/resources/packed_scene.cpp

```diff
@@ -435,24 +411,22 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 
 	for (const DeferredNodePathProperties &dnp : deferred_node_paths) {
 		// Replace properties stored as NodePaths with actual Nodes.
-		Node *other = dnp.base->get_node_or_null(dnp.path);
-
-		const String string_property = dnp.property;
-		if (string_property.contains("/indices/")) {
-			// For properties with "/indices/", the replacement takes place inside an Array.
-			const String base_property = string_property.get_slice("/", 0);
-			const int index = string_property.get_slice("/", 2).to_int();
-
-			Array array = dnp.base->get(base_property);
-			if (array.size() >= index) {
-				array.push_back(other);
-			} else {
-				array.set(index, other);
+		if (dnp.value.get_type() == Variant::ARRAY) {
+			Array paths = dnp.value;
+			if (Engine::get_singleton()->is_editor_hint()) {
+				// If editor, simply set the original array of NodePaths.
+				dnp.base->set(dnp.property, dnp.value);
+				continue;
 			}
```

Comment:
This breaks `@tool` and should be removed entirely. I had kept it with the second test as I was unsure if normal array of `NodePath` could reach here. This wasn't great but as this was never called it did not have any impact. But without the second check it will always pass in editor and node will never be set in editor.
```suggestion

```

Reply:
Will remove tomorrow!